packages/vue2-jest/lib/utils.js

```diff
@@ -68,23 +70,55 @@ const getBabelOptions = function loadBabelOptions(filename, options = {}) {
   return loadPartialConfig(opts).options
 }
 
+const tsConfigCache = {}
+
 /**
  * Load TypeScript config from tsconfig.json.
  * @param {string | undefined} path tsconfig.json file path (default: root)
  * @returns {import('typescript').TranspileOptions | null} TypeScript compilerOptions or null
  */
 const getTypeScriptConfig = function getTypeScriptConfig(path) {
-  const tsconfig = loadTsConfigSync(process.cwd(), path || '')
-  if (!tsconfig.path) {
+  if (path in tsConfigCache) {
+    return tsConfigCache[path]
+  }
+
+  ensureRequire('typescript', ['typescript'])
+  const typescript = require('typescript')
+
+  const tsconfigPath = resolveTsConfigSync(process.cwd(), path || '')
+  if (!tsconfigPath) {
     warn(`Not found tsconfig.json.`)
     return null
   }
-  const compilerOptions =
-    (tsconfig.config && tsconfig.config.compilerOptions) || {}
 
-  return {
-    compilerOptions: { ...compilerOptions, module: 'commonjs' }
+  const parsedConfig = typescript.getParsedCommandLineOfConfigFile(
+    tsconfigPath,
+    {},
+    {
+      ...typescript.sys,
+      onUnRecoverableConfigFileDiagnostic: e => {
+        const errorMessage = typescript.formatDiagnostic(e, {
+          getCurrentDirectory: () => process.cwd(),
+          getNewLine: () => `\n`,
+          getCanonicalFileName: file => file.replace(/\\/g, '/')
+        })
+        warn(errorMessage)
+      }
+    }
+  )
+
+  const compilerOptions = parsedConfig ? parsedConfig.options : {}
+
+  const transpileConfig = {
+    compilerOptions: {
+      ...compilerOptions,
+      module: typescript.ModuleKind.CommonJS
```

Comment:
Note: As in the old code, we override the module with `commonjs`. I changed this to use the enum instead of a string literal because it failed to compile in my typescript version of this code. The expected value of `CompilerOptions['module']` is `ModuleKind`, but it does seem to accept the string literals at runtime.